stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -275,6 +275,18 @@ function eigen(A::AbstractMatrix{T}; permute::Bool=true, scale::Bool=true, sortb
     isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
     return eigen!(AA; permute=permute, scale=scale, sortby=sortby, jvl=jvl, jvr=jvr, jce=jce, jcv=jcv)
 end
+function eigen(A::AbstractMatrix{Float16}; permute::Bool=true, scale::Bool=true, sortby::Union{Function,Nothing}=eigsortby)
```

Comment:
I think we need to allow `Complex{Float16}` here as well. The function body should work as is in both cases. But we should have a test case for the case when eigenvalues of a real matrix are complex, and perhaps another test for `complex(A)` where `A` is real with complex eigenvalues.